### PR TITLE
librdkafka: add version 2.5.0

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **librdkafka/2.5.0**

#### Motivation
There are several new features in 2.5.0.

#### Details
https://github.com/confluentinc/librdkafka/compare/v2.4.0...v2.5.0#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by dotfield — COMMENTED** [review]
Under options (around lines 24-31) there is no option for "snappy". This has been a problem in 2.3.0 and 2.4.0 too. It should be added with the others and have WITH_SNAPPY set to the value.

In reality you should also make SYSLOG configurarable too with these options although we do not use this option. Still, it is an option of the librdkafka build and should be configurable.

**Comment by toge** [comment]
@dotfield 
Thank you for your comment.
I try to fix your point.

**Comment by toge** [comment]
@dotfield 
I found `WITH_SNAPPY` is alwasy set 1 by `config.h` and snappy.h and snappy.c are included upstream repository.

https://github.com/confluentinc/librdkafka/blob/v2.5.0/packaging/cmake/config.h.in#L35

Do you want to add `snappy` option to disable snappy?

**Comment by dotfield** [comment]
> @dotfield I found `WITH_SNAPPY` is alwasy set 1 by `config.h` and snappy.h and snappy.c are included upstream repository.
> 
> https://github.com/confluentinc/librdkafka/blob/v2.5.0/packaging/cmake/config.h.in#L35
> 
> Do you want to add `snappy` option to disable snappy?

It is probably better to set the flag by default to True. That way it should not "break" anyone who has been using it before. Although you have put WITH_SNAPPY=1 in the config.h file it is still set to 0 in the cmake options created which leads to a build warning. It should be consistent, and although it appears that config.h is picked up by rd.h and therefore everything before it hits code checks like in rdkafka_msgset_reader.c, it is still better this is set as an option and is consistent.

The alternative would be to set it to True in conanfile.py (without a configurable option) so it gets picked up by cmake too.

**Comment by toge** [comment]
@dotfield 
Thanks for your advice.
I will add a definition to conanfile.py that sets WITH_SNAPPY to True.

Since it would require modification of config.h to disable the use of SNAPPY depending on the option, I think that it is appropriate to modify on upstream.

**Review by perseoGI — APPROVED** [review]
LGTM

**Review by uilianries — COMMENTED** [review]
Thank you for your PR! Some important points related to this version.

- Google snappy is vendorized in the project, it's possible to decouple, but it would not worth now. The same is available since version 1.9.2, which will not break users using that version: https://github.com/confluentinc/librdkafka/blob/v1.9.2/src/snappy.c

- Would suggest removing syslog option for msvc in config_options. The syslog is available in unix system, but not in Windows. It may work with msys, but probably will fail with visual studio.

**Comment by toge** [comment]
@uilianries 
Thank you for your advice.
I disabled syslog options on msvc.

File: recipes/librdkafka/all/conanfile.py [hunk 1]
```diff
@@ -98,6 +104,8 @@ def generate(self):
         tc.variables["WITH_SASL"] = self.options.sasl
         tc.variables["ENABLE_LZ4_EXT"] = True
         tc.variables["WITH_CURL"] = self.options.curl
+        tc.variables["WITH_SNAPPY"] = True
+        tc.preprocessor_definitions["WITH_SYSLOG"] = "1" if self.options.syslog else "0"
```
Comment (uilianries) [inline recipes/librdkafka/all/conanfile.py:108]:
```suggestion
        tc.preprocessor_definitions["WITH_SYSLOG"] = "1" if self.options.get_safe("syslog") else "0"
```

**Review by uilianries — APPROVED** [review]
LGTM. I tested with syslog active on Linux, and built without errors. Plus, it does not forward an ifdef WITH_SYSLOG in public headers. Full build log:

[librdkafka-2.5.0-syslog-linux.log](https://github.com/user-attachments/files/16466278/librdkafka-2.5.0-syslog-linux.log)

Nice job!

**Review by AbrilRBS — APPROVED** [review]
Thankss!

**Review by perseoGI — APPROVED** [review]

**Pull request merged**